### PR TITLE
add property `cdx:reproducible`

### PR DESCRIPTION
**Pull request by jkowalleck**
Purpose: flag a SBOM document or parts of it as reproducible.
reproducible SBOMs usually omit time-and random-based information, and might render elements in a reproducible order.

some implementations that can generate “reproducible” BOMs, by omitting time- and random-based values, ordering elements, and so on already exist. 

- https://github.com/CycloneDX/cyclonedx-maven-plugin
- https://github.com/CycloneDX/cyclonedx-php-composer
- https://github.com/CycloneDX/cyclonedx-webpack-plugin
- https://github.com/CycloneDX/cyclonedx-node-npm

caused by https://github.com/CycloneDX/cyclonedx-property-taxonomy/pull/69#issuecomment-1667631922

### CODE REVIEW

File: cdx.md [hunk 1]
```diff
@@ -1,7 +1,13 @@
-## `cdx` Namespace Taxonomy
+# `cdx` Namespace Taxonomy
+
+_Boolean value_ are `true` or `false`. Case sensitive.
+
+| Property | Description |
+| -------- | ----------- |
+| `cdx:reproducible` | Whether the CycloneDX document has been generated in a reproducible manner: if so, then time- or random-based values are omitted, and elements should be sorted. _Boolean value_. May appear once. |
```
Comment (hboutemy) [inline cdx.md:7]: "elements should be sorted": "elements order should be reproducible"
to make clear that it's not about actively sorting

Comment (jkowalleck) [reply cdx.md:7]: i know of multiple implementations actively sorting items, so the BOM is effectively reproducible. 

i guess you are right, lets describe the result, not the workflow to achieve it.
i will change this soon.

Comment (jkowalleck) [reply cdx.md]: changes as you suggested.

**Review by hboutemy — APPROVED** [review]

**Comment by jkowalleck** [comment]
@stevespringett could I get your opinion on this proposed property?

File: cdx.md [hunk 2]
```diff
@@ -1,7 +1,13 @@
-## `cdx` Namespace Taxonomy
+# `cdx` Namespace Taxonomy
+
+_Boolean value_ are `true` or `false`. Case sensitive.
+
+| Property | Description |
+| -------- | ----------- |
+| `cdx:reproducible` | Whether the CycloneDX document has been generated in a reproducible manner: if so, then time- or random-based values MUST be omitted, and elements order SHOULD be reproducible. _Boolean value_. May appear once. |
```
[the remark this answers is not in the document]
Comment (jkowalleck) [reply cdx.md:7]: what about it? 
all build-time information that is related to `time` is optional in CycloneDX, so just omit it.

If you going to fake/pretend timestamps, then why bother?

[the remark this answers is not in the document]
Comment (mathstuf) [reply cdx.md:7]: The idea of it is to say "the date is when the source artifact was made/published/whatever". I suppose just omitting it here is fine, but not having to scrounge for some date information elsewhere would be useful I expect…

[the remark this answers is not in the document]
Comment (jkowalleck) [reply cdx.md:7]: could you open a [new issue](https://github.com/CycloneDX/cyclonedx-property-taxonomy/issues/new) for the topic, so we could discuss further? 
please describe why you need the timestamp, for what you use it, and what you want to artificially base it of, and what role the "reproducible" flag plays in all that.

**Pull request merged**